spanner/spannertest/integration_test.go

```diff
@@ -439,6 +441,358 @@ func TestIntegration_SpannerBasics(t *testing.T) {
 	if !reflect.DeepEqual(ages, wantAges) {
 		t.Errorf("Query with IN UNNEST gave wrong ages: got %+v, want %+v", ages, wantAges)
 	}
+	t.Run("Read all from index", func(t *testing.T) {
+		cases := []struct {
+			indexName string
+			want      []int64
+		}{
+			{
+				indexName: "AgeIndexAsc",
+				want:      []int64{0, 0, 35, 49, 102},
+			},
+			{
+				indexName: "AgeIndexDesc",
+				want:      []int64{102, 49, 35, 0, 0},
+			},
+			{
+				indexName: "AgeIndexStoring",
+				want:      []int64{0, 0, 35, 49, 102},
+			},
+		}
+
+		for _, tc := range cases {
+			ages := ages[:0]
+			if err := client.Single().ReadUsingIndex(ctx, tableName, tc.indexName, spanner.AllKeys(), []string{"Age"}).Do(func(row *spanner.Row) error {
+				var age spanner.NullInt64
+				if err := row.Column(0, &age); err != nil {
+					return err
+				}
+				ages = append(ages, age.Int64) // zero for NULL
+				return nil
+			}); err != nil {
+				t.Fatalf("Getting ages using index: %v", err)
+			}
+			if !reflect.DeepEqual(ages, tc.want) {
+				t.Fatalf("Read index %s gave wrong ages: got %+v, want %+v", tc.indexName, ages, tc.want)
+			}
+		}
+	})
+	t.Run("Read all from index desc", func(t *testing.T) {
+		ages := ages[:0]
+		if err := client.Single().ReadUsingIndex(ctx, tableName, "AgeIndexDesc", spanner.AllKeys(), []string{"Age"}).Do(func(row *spanner.Row) error {
+			var age spanner.NullInt64
+			if err := row.Column(0, &age); err != nil {
+				return err
+			}
+			ages = append(ages, age.Int64) // zero for NULL
+			return nil
+		}); err != nil {
+			t.Fatalf("Getting ages using index: %v", err)
+		}
+		wantAges = []int64{102, 49, 35, 0, 0}
+		if !reflect.DeepEqual(ages, wantAges) {
+			t.Fatalf("Read index gave wrong ages: got %+v, want %+v", ages, wantAges)
+		}
+	})
+	t.Run("Read ranges from index", func(t *testing.T) {
```

Comment:
These test cases (or in a separate test function) should be extended with cases for:
* Reading single keys (one or more)
* The combination of reading single keys + ranges
* Limit in combination with the above.